include/bluetooth/mesh/access.h

```diff
@@ -281,7 +281,14 @@ struct bt_mesh_model_pub {
 	 *  contain a valid publication message before calling the
 	 *  bt_mesh_model_publish() API or after the publication's
 	 *  @ref bt_mesh_model_pub.update callback has been called
-	 *  and returned success.
+	 *  and returned success. The buffer must be created outside
+	 *  of function context, i.e. it must not be on the stack.
```

Comment:
Could be a static inside the function context? I suggest just keeping the fact that it must not be in the call stack

Reply:
The thing is, I'm not sure you can ever create a static net_buf_simple using the NET_BUF_SIMPLE() macro. The reason is that putting static in front of the declaration only makes the pointer static, but the macro expands to a reference to an anonymous net_buf_simple object, and that object doesn't get the static qualifier.

Reply:
Here's what the macro expands to:
```
#define NET_BUF_SIMPLE(_size)                        \                          
        ((struct net_buf_simple *)(&(struct {        \                          
                struct net_buf_simple buf;           \                          
                u8_t data[_size] __net_buf_align; \                             
        }) {                                         \                          
                .buf.size = _size,                   \                          
        }))
```

Reply:
So right now, I think the safest recommendation is to ask users to declare the buffer outside of function context. That said, we could consider creating a NET_BUF_SIMPLE_STATIC macro that puts the static qualifier in the right place. However, we also need to consider the upcoming refactoring where the data portion is split behind a separate pointer instead of implicitly following the buffer meta-data struct.

Reply:
Regarding the net_buf API redesign, currently in that world (my not yet published work) the macro looks like this:
```
#define NET_BUF_SIMPLE_DEFINE(_name, _size)  \                                  
        u8_t ##_name_data[_size];            \                                  
        struct net_buf_simple _name = {      \                                  
                .data   = ##_name_data,      \                                  
                .len    = 0,                 \                                  
                .size   = _size,             \                                  
                .__buf  = ##_name_data,      \                                  
        };      
```

So there'd then have to be a `NET_BUF_SIMPLE_STATIC_DEFINE()` to make both of those expanded variables static. It's noteworthy that this will unfortunately no-longer be usable in the same inline-style for initializing struct members. However, we're getting a bit ahead of ourselves here :)

Reply:
I see, thank you for the explanation Johan. Given all the input you have given me and the upcoming rework, on top of the fact that it is really not possible simply declare a static version of a simple net_buf, I am happy with the current description.